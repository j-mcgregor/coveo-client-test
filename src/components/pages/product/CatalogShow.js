/* eslint-disable operator-linebreak */
import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import stringToKey from '../../../utils/createKey';

const CatalogShow = ({ fetch, products }) => {
  const { type } = useParams();

  const tiles =
    products &&
    products.map((product, i) => (
      <div className="column col-3" key={stringToKey(product.name, i)}>
        <div className="card" style={{ height: '13rem', margin: '1rem 0' }}>
          <div className="card-image">
            <img src="img/osx-el-capitan.jpg" className="img-responsive" />
          </div>
          <div className="card-header">
            <div className="card-title h5">{product.name}</div>
            <div className="card-subtitle text-gray">
              {product.city}, {product.state}
            </div>
          </div>
          <div className="card-body">...</div>
          <div className="card-footer">
            <a href={product.website_url}>
              {' '}
              <i className="fas fa-globe" />{' '}
            </a>
          </div>
        </div>
      </div>
    ));

  useEffect(() => {
    fetch(type);
  }, []);

  return (
    <div className="container grid-lg">
      <div className="hero bg-gray">
        <div className="hero-body">
          <h3>{type}</h3>
        </div>
      </div>
      <div className="columns">{tiles}</div>
    </div>
  );
};

export default CatalogShow;
