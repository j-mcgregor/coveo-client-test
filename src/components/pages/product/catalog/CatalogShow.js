/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { useParams } from 'react-router-dom';
import GoogleMapReact from 'google-map-react';
import S from '../../../../static/styles';
import marker from '../../../../static/images/marker.png';
import config from '../../../../../config';

const formatPhone = number => {
  const cleaned = `${number}`.replace(/\D/g, '');
  const match = cleaned.match(/^(\d{3})(\d{3})(\d{4})$/);
  if (match) {
    return `(${match[1]}) ${match[2]}-${match[3]}`;
  }
  return null;
};

const Marker = () => (
  <img
    src={marker}
    alt="Marker"
    width="40"
    style={{ transform: 'translate(-50%, -50%)' }}
  />
);

const CatalogShow = ({ selected, closeBanner }) => {
  const { type } = useParams();
  let banner;

  if (type === 'breweries' && !selected.wine) {
    const {
      name,
      brewery_type,
      country,
      street,
      city,
      state,
      phone,
      website_url,
      longitude,
      latitude
    } = selected;
    banner = (
      <div className="container grid-lg">
        <div className="columns">
          <div className="column col text-right">
            <i
              className="fas fa-times"
              onClick={closeBanner}
              role="presentation"
            />
          </div>
        </div>
        <div className="columns">
          <div className="column col">
            <h2>{name}</h2>
            <h6>Size: {brewery_type}</h6>
            {phone && (
              <div>
                <small>
                  <i className="fas fa-phone-alt" /> {formatPhone(phone)}
                </small>
              </div>
            )}
            {website_url && (
              <div>
                <small>
                  <a
                    href={website_url}
                    target="_blank"
                    rel="noopener noreferrer"
                  >
                    <i className="fas fa-globe" /> website
                  </a>
                </small>
              </div>
            )}
          </div>
          <div className="column col text-right">
            <div className="column col">
              <small>
                <div>{street}</div>
                <div>{city}</div>
                <div>{state}</div>
                <div>{country}</div>
              </small>
              {latitude && longitude ? (
                <div style={{ height: '200px', width: '100%' }}>
                  <GoogleMapReact
                    bootstrapURLKeys={{
                      key: config.GOOGLE_MAPS_API_KEY
                    }}
                    defaultCenter={{
                      lat: parseFloat(latitude),
                      lng: parseFloat(longitude)
                    }}
                    defaultZoom={14}
                  >
                    <Marker
                      lat={parseFloat(latitude)}
                      lng={parseFloat(longitude)}
                    />
                  </GoogleMapReact>
                </div>
              ) : (
                <small>
                  <a
                    href={`http://maps.google.com/?q=${`${street},${city},${state},${country}`}`}
                    target="_blank"
                    rel="noopener noreferrer"
                  >
                    See on map
                  </a>
                </small>
              )}
            </div>
          </div>
        </div>
      </div>
    );
  }

  if (type === 'wines' && !selected.brewery_type) {
    const {
      wine,
      country,
      regions,
      vintage,
      wine_slug,
      score,
      confidence_index,
      journalist_count
    } = selected;
    banner = (
      <div className="container grid-lg">
        <div className="columns">
          <div className="column col text-right">
            <i
              className="fas fa-times"
              onClick={closeBanner}
              role="presentation"
            />
          </div>
        </div>
        <div className="columns">
          <div className="column col">
            <h2>{wine}</h2>
            <div>
              {regions.length ? `${regions[0]}, ` : ''}
              {country}
            </div>
            <small>Vintage: {vintage}</small>
          </div>
          <div className="column col-3 text-right">
            <div className="columns">
              <div className="column col text-right">
                Average Score ({journalist_count}):{' '}
              </div>
              <div className="column col-3">{score}%</div>
            </div>
            <div className="columns">
              <div className="column col text-right">Confidence Index: </div>
              <div className="column col-3">{confidence_index}</div>
            </div>
            <small>
              <a
                href={`http://maps.google.com/?q=${wine_slug}`}
                target="_blank"
                rel="noopener noreferrer"
              >
                See on map
              </a>
            </small>
          </div>
        </div>
      </div>
    );
  }

  return (
    <S.CatalogShow className="container" style={{ marginTop: '36px' }}>
      {banner}
    </S.CatalogShow>
  );
};

const mapStateToProps = state => ({
  selected: state.catalog.selected
});

CatalogShow.propTypes = {
  selected: PropTypes.shape({}),
  closeBanner: PropTypes.func
};

CatalogShow.defaultProps = {
  selected: {},
  closeBanner: () => {}
};

export default connect(mapStateToProps)(CatalogShow);
