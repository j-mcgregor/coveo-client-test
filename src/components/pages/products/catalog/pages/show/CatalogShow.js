/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';
import { useParams } from 'react-router-dom';
import S from '../../../../../../static/styles';
import BreweryBanner from './banners/Brewery';
import WineBanner from './banners/Wine';

const CatalogShow = ({ selected, closeBanner }) => {
  const { type } = useParams();
  let banner;

  if (type === 'breweries' && !selected.wine) {
    banner = <BreweryBanner {...selected} closeBanner={closeBanner} />;
  }

  if (type === 'wines' && !selected.brewery_type) {
    banner = <WineBanner {...selected} closeBanner={closeBanner} />;
  }

  return (
    <S.CatalogShow className="container grid-lg" style={{ marginTop: '36px' }}>
      {banner}
    </S.CatalogShow>
  );
};

CatalogShow.propTypes = {
  selected: PropTypes.shape({}),
  closeBanner: PropTypes.func
};

CatalogShow.defaultProps = {
  selected: {},
  closeBanner: () => {}
};

export default CatalogShow;
