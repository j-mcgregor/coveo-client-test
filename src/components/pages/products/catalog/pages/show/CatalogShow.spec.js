/* eslint-disable react/prop-types */
import React from 'react';
import { shallow } from 'enzyme';
import CatalogShow from './CatalogShow';

jest.mock('react-router', () => ({
  useParams: jest.fn().mockReturnValue({ type: 'breweries' })
}));

describe('CatalogShow', () => {
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      closeBanner: jest.fn(),
      selected: {}
    };
  });

  it('should render', () => {
    const wrapper = shallow(<CatalogShow {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
