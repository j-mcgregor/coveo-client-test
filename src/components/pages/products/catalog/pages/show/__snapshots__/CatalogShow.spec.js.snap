// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogShow should render 1`] = `ShallowWrapper {}`;
